eventlog/tests.py

```diff
@@ -0,0 +1,71 @@
+from django.contrib.auth.models import User
+from django.test import TestCase
+from django.utils import timezone
+
+from roster.models import Site, Period, ClusiveUser
+from eventlog.models import Session, Event
+
+# TODO copied from roster tests. Is there a central place that some of this setup could be done?
+def set_up_test_sites():
+    Site.objects.create(name="CAST Collegiate", city="Wakefield", state_or_province="MA", country="USA").save()
+
+def set_up_test_periods():
+    cast_collegiate = Site.objects.get(name="CAST Collegiate")
+    Period.objects.create(name="Universal Design For Learning 101", site=cast_collegiate).save()
+
+def set_up_test_users():
+    user_1 = User.objects.create_user(username="user1", password="password1")
+    user_1.save()
+    cuser_1 = ClusiveUser.objects.create(anon_id="Student1", user=user_1, role='ST')
+    cuser_1.periods.add(Period.objects.get(name="Universal Design For Learning 101"))
+    cuser_1.save()
+
+class EventlogTestCase(TestCase):
+    def setUp(self):
+        set_up_test_sites()
+        set_up_test_periods()
+        set_up_test_users()
+
+    def test_session_created(self):
+        login = self.client.login(username='user1', password='password1')
+        self.assertTrue(login)
+        session = Session.objects.all()[0]
+        self.assertEquals(len(session.id), 36, "Session should get uuid assigned")
+        self.assertEquals(session.user, ClusiveUser.objects.all()[0], "Session should connect to the user")
+        self.assertEquals(session.startedAtTime.date(), timezone.now().date(), "Should set reasonable session start time")
+        self.assertIsNotNone(session.userAgent)
+
+    def test_session_closed(self):
+        login = self.client.login(username='user1', password='password1')
+        self.assertTrue(login)
+        self.client.logout()
+        session = Session.objects.all()[0]
+        self.assertEquals(session.endedAtTime.date(), timezone.now().date(), "Should set reasonable session end time")
```

Comment:
The default session timeout looks to be two weeks: https://docs.djangoproject.com/en/dev/ref/settings/#session-cookie-age - we should be able to adjust this in `settings.py`

It occurs to me that we should also verify that a session timeout sends the same signal as a manual logout, if we want to be able to record it as one.

Reply:
By default Django doesn't seem to have a notion of a session timing out.
May need add on such as https://pypi.org/project/django-session-idle-timeout/ ?